feat!: aztec binary

spypsy opened the pull request:
Closes #3552 
Closes #3811 

- New `aztec` package, replaces `aztec-sandbox`
- New `aztec` CLI to run Aztec infra (sandbox, node, sequencer, archiver, p2p-bootstrap)
Draft for now to discuss tooling naming

# Checklist:
Remove the checklist to signal you've completed it. Enable auto-merge if the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g. cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to relevant issues (if any exist).

spypsy committed:
WIP 'aztec' CLI & renaming sandbox

spypsy committed:
finish renaming

spypsy committed:
add standalone archiver

spypsy committed:
add p2p bootstrap node

spypsy committed:
command descriptions

spypsy committed:
merge with master

spypsy committed:
typing fixes

spypsy committed:
update scripts

spypsy committed:
remove old sandbox bin

spypsy committed:
add 'aztec cli' call

spypsy committed:
merge with master

spypsy committed:
fix compose files

spypsy committed:
fix ports

spypsy committed:
merge with master

spypsy committed:
fix typo

spypsy committed:
fix more imports

spypsy committed:
merge with master

spypsy committed:
fix build

spypsy committed:
fix imports

spypsy commented on yarn-project/aztec.js/src/index.ts:
```diff
@@ -48,7 +48,7 @@ export {
 } from './utils/index.js';
 
-export { createPXEClient } from './pxe_client.js';
+export { createPXEClient, createAztecNodeClient, createArchiverClient } from './rpc_clients/index.js';
```
if we export `createPXEClient` from aztec.js, doesn't it make sense to do the same for Node & Archiver?
Or you think these shouldn't be exposed to external users at all and just have them in their respective packages

PhilWindle replied:
No, they don't belong on the public api. `createPXEClient` enables a dev to programatically comminicate with a PXE. They shouldn't need to communicate directly with either the node or the arcihiver.

They should just be in their respective packages.

spypsy committed:
remove openDB fn from kv-store

spypsy committed:
undo formatting

spypsy committed:
move rpc clients

spypsy committed:
multi-class rpc server

spypsy committed:
WIP using multi json rpc server

spypsy committed:
JSON RPC fixes

spypsy committed:
Merge branch 'master' into spy/aztec-package

spypsy committed:
debug logging

spypsy committed:
update lmdb

alexghr committed:
fix: pxe notes crashing spuriously (#4049)

This PR refactors how the PXE stores notes. They're now stored by their
index in the note hash tree which should provide both uniqueness and a
sorted order for notes.

This gets rid of the "nullifiedNotes" set so reading and removing notes
should be cleaner and more efficient.

spypsy committed:
multi-jsonRPC server fixes

spypsy committed:
export types

spypsy committed:
Merge branch 'master' into spy/aztec-package

spypsy committed:
update tests

spypsy committed:
fix formatting

spypsy committed:
remove json-rpc logs

Commit:
fix: notes database

spypsy committed:
skip return conversion as well in multi rpc

spypsy committed:
merge with master

spypsy committed:
update boxes yarn.lock

Commit:
fix: give mock Notes an index

spypsy committed:
fix blank box webpack

spypsy committed:
fix json-rpc conversion

spypsy committed:
Merge branch 'master' into spy/aztec-package

spypsy committed:
update terraforms

spypsy committed:
Merge branch 'master' into spy/aztec-package

spypsy committed:
remove openDb util

spypsy committed:
yarn prepare

spypsy committed:
update sandbox reference

spypsy commented on yarn-project/deploy_npm.sh:
```diff
@@ -99,3 +99,4 @@ deploy_package p2p
 deploy_package world-state
 deploy_package sequencer-client
 deploy_package aztec-node
+deploy_package aztec
```
hmm I guess not since we no longer recommend the npm packages to users

spypsy committed:
Addressing PR comments

spypsy committed:
Remove lodash.pick from aztec package

spypsy committed:
allow 0x private keys in aztec-cli

spypsy committed:
Fix archiver client + contract address requirements

spypsy committed:
namespace rpc server by default, remove unused 'useApi' option

spypsy committed:
'aztec sandbox' starts the full sandbox

spypsy committed:
remove 'sandbox' command

spypsy committed:
Merge branch 'master' into spy/aztec-package

PhilWindle commented on yarn-project/aztec/src/cli/cli.ts:
```diff
@@ -0,0 +1,196 @@
+      // Start Aztec Node
+      if (options.node) {
+        // get env vars first
+        const aztecNodeConfigEnvVars = getNodeConfigEnvVars();
```
Can we break these out into separate methods to make it easier to read?

PhilWindle commented on yarn-project/aztec/src/cli/texts.ts:
```diff
@@ -0,0 +1,70 @@
+
+export const cliTexts = {
+  node:
+    'Starts Aztec Node with options.\n' +
```
I think this should have a data directory shouldn't it? For storing transactions and world state at least.

spypsy committed:
PR Fixes

spypsy committed:
fix typo

spypsy commented:
tagging a devrel @signorecello

spypsy committed:
Merge branch 'master' into spy/aztec-package

spypsy committed:
update lmdb store method

spypsy committed:
remove unused packages from aztec

spypsy committed:
merge with master
